tests/integration/istio-injection/istioinjection/main_test.go

```diff
@@ -0,0 +1,74 @@
+package istioinjection
+
+import (
+	"fmt"
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	namespaceNameRoot = "istio-injection-tests"
+	noSidecar         = 1
+	hasSidecar        = 2
+)
+
+type TestSuite struct {
+	k8sClient *kubernetes.Clientset
+	namespace string
+}
+
+func TestIstioInjection(t *testing.T) {
+	namespace := fmt.Sprintf("%s-%s", namespaceNameRoot, generateRandomString(8))
+	if namespace == "" {
+		log.Fatal("Namespace not set.")
+	}
+
+	testSuite := &TestSuite{namespace: namespace}
+	testSuite.initK8sClient()
+
+	defer testSuite.deleteNamespace()
+	if err := testSuite.createNamespace(); err != nil {
+		panic(err)
+	}
+
+	cases := []struct {
+		description                   string
+		disableInjectionForNamespace  bool
+		disableInjectionForDeployment bool
+		expectedResult                int
+	}{
+		{"no injection flag for both namespace and deployment", false, false, hasSidecar},
+		{"deployment injection is disabled", false, true, noSidecar},
+		{"namespace injection is disabled", true, false, noSidecar},
+		{"namespace and deployment injections are disabled", true, true, noSidecar},
+	}
+
+	for _, c := range cases {
+		t.Run(c.description, func(t *testing.T) {
+			testSuite.disableInjectionForNamespace(c.disableInjectionForNamespace)
+			testID := generateRandomString(8)
+
+			_, err := testSuite.createDeployment(testID, c.disableInjectionForDeployment)
+			if err != nil {
+				log.Errorf("Cannot create deployment '%s': %v", testID, err)
```

Comment:
Maybe we should stop the program here, too? `log.Errorf` doesn't do that, it only logs the message. Consider using `log.Panic`.